src/common/keygen.rs

```diff
@@ -0,0 +1,84 @@
+//! Contains functions used to generate scrambled alphabets from a key.
+//!
+
+use std::ascii::AsciiExt;
+use super::alphabet::{find_position, get_letter};
+
+/// Generates a scrambled alphabet using a key. Repeated letters in the key are ignored.
+/// e.g. A key of `alphabet` will produce the result `alphbetcdfgijkmnoqrsuvwxyz`.
+///
+/// Will return Err if a non-alphabetic symbol is present in the key.
+pub fn generate_keyed_alphabet(key: &str, make_uppercase: bool) -> Result<String, &'static str> {
+
+    // A String to store our new keyed alphabet.
+    let mut keyed_alphabet = String::new();
+
+    // Loop over the key and add each unique letter to the keyed alphabet.
+    for c in key.chars() {
+        match find_position(c) {

```

Comment:
To make it uniform across the codebase (and more easy to read) could you change this to `alphabet::find_position(c)`, same with `get_letter`